rust/src/config.rs

```diff
@@ -142,3 +158,74 @@ impl ARCH {
             .contains(&arch.to_ascii_lowercase().as_str())
     }
 }
+
+pub struct StringKey<'a>(pub Vec<&'a str>, pub &'a str);
+
+impl StringKey<'_> {
```

Comment:
Every time StringKey is used it's just used to call get_key() on it. It might make more sense to just convert it to a function eg

```rust
pub struct Keys; 
impl Keys {
  get_string_key(key: &str, default: &str){...}
  }